android-json-form-wizard/src/main/java/com/vijay/jsonwizard/validators/edittext/LengthValidator.java

```diff
@@ -18,8 +18,7 @@ public LengthValidator(String errorMessage, int minLength, int maxLength) {
 
     @Override
     public boolean isValid(CharSequence charSequence, boolean isEmpty) {
```

Comment:
What if we used the `isEmpty` boolean to return `true` since it is a parameter passed by the super to tell whether the text is empty?

Alternatively we can update the constructor to include an `isRequired` flag and then check that to make the validation check. That way it is also invalid if the _isRequired_ is **true** and char sequence is empty

Reply:
Actually I think we can update the constructor to include an `isRequired` flag and then check that to make the validation check. That way it is also invalid if the _isRequired_ is **true** and char sequence is empty

Reply:
This is noted.
I think the `isEmpty` flag sent to the validator is redundant since we check for the field's length when doing the other checks for min and max length validators. `isEmpty` flag may also mean different things for mandatory or non-mandatory fields, or in case of a minimum length greater than zero.

Reply:
Also bundling a required validator for this may have unintended results when forms are not properly defined.

Reply:
The _isRequired_ field could then be populated to `true` if in the json form filed _**it exists**_ and **_it is set to true_** otherwise `false`

Reply:
The _isRequired_ field could then be populated to `true` if in the json form field _**it exists**_ and **_it is set to true_** otherwise `false`